### PR TITLE
Run4-hgx261 Bug fix for the v14 geometry of HGCal

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Bug fix for the v14 geometry of HGCal

#### PR validation:

Tested using runTheMatrix workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
@fabiocos changes in phase 2 workflows seem to arise from fluctuations in the FTL RecHit time value:
all_OldVSNew_TTbar14TeV2026D49wf23234p0c_FTLRecHitsSorted_mtdRecHits_FTLEndcap_RECO_obj_obj_time
is this expected?

**Comment by cvuosalo** [comment]
+1

**Comment by bsunanda** [comment]
@kpedro88 this is not from this PR

**Comment by kpedro88** [comment]
@bsunanda sorry, posted on the wrong PR! will move that comment

File: SimG4CMS/Calo/src/HGCalNumberingScheme.cc [hunk 1]
```diff
@@ -50,6 +51,18 @@ uint32_t HGCalNumberingScheme::getUnitID(int layer, int module, int cell, int iz
       hgcons_.waferFromPosition(xx, pos.y(), layer, waferU, waferV, cellU, cellV, waferType, wt);
     }
     if (waferType >= 0) {
+      if (mode_ == HGCalGeometryMode::Hexagon8File) {
+        int type = hgcons_.waferType(layer, waferU, waferV, true);
+        if (type != waferType)
```
Comment (kpedro88) [inline SimG4CMS/Calo/src/HGCalNumberingScheme.cc:56]: is this if statement only supposed to apply to the `edm::LogVerbatim` line below? If so, I would put the if statement inside the `ifdef`.

File: SimG4CMS/Calo/src/CaloSD.cc [hunk 2]
```diff
@@ -107,6 +107,7 @@ CaloSD::~CaloSD() {}
 
 G4bool CaloSD::ProcessHits(G4Step* aStep, G4TouchableHistory*) {
   NaNTrap(aStep);
+  ignoreReject = false;
```
Comment (kpedro88) [inline SimG4CMS/Calo/src/CaloSD.cc:110]: Does this override the `ignoreRejection()` setter that is called elsewhere? If so, this line should probably be removed.

File: SimG4CMS/Calo/src/HGCalNumberingScheme.cc (hunk 1 above)
Comment (bsunanda) [reply SimG4CMS/Calo/src/HGCalNumberingScheme.cc:56]: Thanks for pointing out this. I am correcting for that

File: SimG4CMS/Calo/src/CaloSD.cc (hunk 2 above)
Comment (bsunanda) [reply SimG4CMS/Calo/src/CaloSD.cc:110]: This is the initialization of the flag ignoreReject. For certain cases it is called by some specific case - for HGCal it is done by setDetUnitId method.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
+upgrade

**Comment by kpedro88** [comment]
please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

**Comment by bsunanda** [comment]
@civanch @kpedro88 Could you please approve this PR?

**Comment by kpedro88** [comment]
+upgrade

**Comment by civanch** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**